src/fluree/store/file.cljc

```diff
@@ -0,0 +1,51 @@
+(ns fluree.store.file
+  (:refer-clojure :exclude [read])
+  (:require [fluree.crypto :as crypto]
+            [fluree.db.util.filesystem :as fs]
+            [fluree.store.proto :as store-proto]
+            [fluree.store.util :as store-util]))
+
+(defn file-write
+  [storage-path k v {:keys [content-address?] :as opts}]
+  (when (not (store-util/hashable? v))
+    (throw (ex-info "Must serialize v before writing to FileStore."
+                    {:storage-path storage-path
+                     :k            k
+                     :v            v
+                     :opts         opts})))
+  (let [hash (crypto/sha2-256 v)
+        k*   (if content-address?
```

Comment:
we shouldn't have a content-address? option here, and we should also move away of embedding control flow in predicate? options in general. this is the point of protocols and records. the storage api needs to be content addressed, and if there's a use case that cannot be content addressed, then that use case should be supported by a different api because they are performing fundamentally different actions.

Reply:
Same as with the proto point, I'd like to defer this to the IPFS conn work in the interest of shipping https://github.com/fluree/core/issues/57 and https://github.com/fluree/core/issues/58

If you don't agree, I can absolutely do it as part of this, though.